### PR TITLE
postgresql modules: use postgres.exec_sql instead of __exec_sql methods

### PR DESCRIPTION
**Pull request by Andersson007**
##### SUMMARY
postgresql modules: use postgres.exec_sql instead of __exec_sql methods

##### ISSUE TYPE

- Bugfix Pull Request

### CODE REVIEW

**Comment by Andersson007** [comment]
ready_for_review

**Comment by andytom** [comment]
LGTM, I wonder if it is worth creating a little mixin class in the future that the different module classes can inherit from.

**Comment by Andersson007** [comment]
@andytom , I though about this.
One developer of core team recently recommended me to watch [Stop writing classes](https://www.youtube.com/watch?v=o9pEzgHorH0) and I decided to implement this (and several previous functions) as a function.
However I don't mind if somebody made the class well (IMO usually more flexible to use composition but I don't min'd if it is inheritance because the case is really simple. If it is implemented really good, why not).
Thank you for the suggestion and reviewing!

**Comment by Andersson007** [comment]
@andytom , please, shipit

**Comment by andytom** [comment]
shipit

File: changelogs/fragments/57674-postgres_modules_use_exec_sql_instead_of_methods.yml
Comment (felixfontein) [inline changelogs/fragments/57674-postgres_modules_use_exec_sql_instead_of_methods.yml]:
```suggestion
  - postgresql modules - use ``module_utils.postgres.exec_sql`` function instead of ``__exec_sql`` method (https://github.com/ansible/ansible/pull/57674)
```

**Review by felixfontein — APPROVED** [review]
LGTM

**Pull request merged**
